lms/djangoapps/discussion_api/api.py

```diff
@@ -751,20 +751,28 @@ def _do_extra_actions(api_content, cc_content, request_fields, actions_form, con
                 raise ValidationError({field: ["Invalid Key"]})
 
 
-def _handle_following_field(form_value, user, cc_content):
+def _handle_following_field(form_value, user, cc_content, api_content):
     """follow/unfollow thread for the user"""
     if form_value:
         user.follow(cc_content)
+        if cc_content["type"] == "thread":
+            api_content["read"] = True
```

Comment:
How do you feel about the following comment?
```
# always return read as True as reasonably accurate shortcut, rather than adding additional processing.
```